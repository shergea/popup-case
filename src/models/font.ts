export default interface Font {
  entities: Array<FontItem>;
  loading: "idle" | "pending" | "succeeded" | "failed";
  expireTime: number;
}

export interface FontItem {
  family: string;
  subsets: string[];
  version: string;
  category: string;
  variants: string[];
}
