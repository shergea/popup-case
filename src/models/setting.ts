export default interface Setting {
  headline: string;
  description: string;
  successMessage: string;
  buttonText: string;
}
