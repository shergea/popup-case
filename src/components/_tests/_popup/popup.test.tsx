import React from "react";
import renderer from "react-test-renderer";
import { FontItem } from "../../../models/font";
import Setting from "../../../models/setting";
import Popup from "../../popup";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <Popup
        setting={
          {
            headline: "HeadLine",
            buttonText: "ButtonText",
            description: "Description",
            successMessage: "successMessage",
            font: "Roboto",
          } as Setting
        }
        fonts={
          [
            {
              family: "ABeeZee",
              subsets: ["latin", "latin-ext"],
              version: "v22",
              category: "sans-serif",
              variants: ["regular", "italic"],
            },
            {
              family: "Abel",
              subsets: ["latin"],
              version: "v18",
              category: "sans-serif",
              variants: ["regular"],
            },
          ] as Array<FontItem>
        }
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
