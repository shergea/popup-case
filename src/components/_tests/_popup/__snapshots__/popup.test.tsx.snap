// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="boxBackground"
  data-test="popup-1"
  style={
    Object {
      "fontFamily": "Roboto",
    }
  }
>
  <div
    className="flex flex-col justify-center items-center bg-white text-center m-auto box"
  >
    <button
      className="buttonClose"
      onClick={[Function]}
      type="button"
    >
      <span>
        +
      </span>
    </button>
    <h1
      className="headline"
      data-test="popup-headline"
    >
      HeadLine
    </h1>
    <h2
      className="description"
      data-test="popup-description"
    >
      Description
    </h2>
    <form
      action="#"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="mt-4"
      >
        <input
          className="border border-gray-300 bg-gray-100 text-base font-medium block w-full p-2.5 placeholder:text-black placeholder:text-base placeholder:font-medium outline-0"
          id="fullName"
          name="fullName"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Your Name"
          type="text"
          value=""
        />
      </div>
      <div
        className="mt-4"
      >
        <input
          className="border border-gray-300 bg-gray-100 text-base font-medium block w-full p-2.5 placeholder:text-black placeholder:text-base placeholder:font-medium outline-0"
          id="email"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Email"
          type="text"
          value=""
        />
      </div>
      <div
        className="mt-4"
      >
        <select
          className="border border-gray-300 bg-gray-100 text-base block w-full p-2.5 font-medium outline-0"
          id="font"
          name="font"
          onBlur={[Function]}
          onChange={[Function]}
          value=""
        >
          <option
            value=""
          >
            Select Font
          </option>
          <option
            value="ABeeZee"
          >
            ABeeZee
          </option>
          <option
            value="Abel"
          >
            Abel
          </option>
        </select>
      </div>
      <button
        className="block w-full p-2.5 mt-5 text-base font-medium text-white buttonGreen"
        type="submit"
      >
        ButtonText
      </button>
    </form>
  </div>
</div>
`;
